Java: CWE-200: Temp directory local information disclosure vulnerability (#4388)
github/codeql: java/ql/src/change-notes/2022-02-04-local-temp-file-or-directory-information-disclosure.md

---
category: newQuery
---

> * A new query titled "Temporary directory Local information disclosure" (`java/local-temp-file-or-directory-information-disclosure`) has been added.

Reviewer: If we change the name of the query, we'd also need to update it here.

> * Two new querys both titled "Temporary directory Local information disclosure" (`java/local-temp-file-or-directory-information-disclosure-path`, `java/local-temp-file-or-directory-information-disclosure-method`) have been added.
>   These queries find uses of APIs that leak potentially sensitive information to other local users via the system temporary directory.
>   This query was originally [submitted as query by @JLLeitschuh](https://github.com/github/codeql/pull/4388).

Reviewer 2: ```suggestion
* A new query titled "Temporary directory Local information disclosure" (`java/local-temp-file-or-directory-information-disclosure`) has been added.
  This query finds uses of APIs that leak potentially sensitive information to other local users via the system temporary directory.
  This query was originally [submitted as query by @JLLeitschuh](https://github.com/github/codeql/pull/4388).
```
